### ISSUE TITLE
Client Certificates not possible when using IIS Hosting

### ISSUE THREAD

**Issue by Viajaz** [Badgerati/Pode#979]
### Describe the Bug
Let the Pode endpoint be:
```powershell
Add-PodeEndpoint -Address '127.0.0.1' -Port $env:ASPNETCORE_PORT -Protocol Http
```
-correctly returns HTTP 401 when a HTTPS request is made without a Client Certificate, however, it always returns HTTP 403 when a HTTPS request, with a Client Certificate, is made to a Pode Route that utilises a Client Certificate Auth Scheme.

The documentation states:
https://github.com/Badgerati/Pode/blob/f6587253f6889e0e2651e33f43c18ddea009eec5/docs/Hosting/IIS.md?plain=1#L303

-however, one cannot use `AllowClientCertificate` with a `Http` endpoint due to:
https://github.com/Badgerati/Pode/blob/edded58be9897b61291a8cfefef688c707140d71/src/Public/Core.ps1#L912-L914

-which I believe means `$WebEvent.Request.AllowClientCertificate` is false, resulting in the Client Certificate Check Middleware for IIS never parsing the  `MS-ASPNETCORE-CLIENTCERT` or `X-ARR-ClientCert` Headers due to:
https://github.com/Badgerati/Pode/blob/533dc05584a1b6156ec98dd773fb52a9893de593/src/Private/Middleware.ps1#L390-L392

### Steps To Reproduce
1. Configure IIS as per IIS Hosting Guide
2. 'Accept' Client Certificates in the SSL Settings of the IIS Site hosting Pode
3. Perform HTTPS Request against Pode Route.

### Expected Behavior
1. For the IIS Middleware to accept and parse `MS-ASPNETCORE-CLIENTCERT` or `X-ARR-ClientCert` when applicable.
2. For the Documentation regarding Client Certificates in an IIS Hosting situation to be correct.

### Screenshots
N/A

### Platform
 - OS: Windows Server 2019 [10.0.17763]
 - Browser: WindowsPowerShell/5.1.19041.1645
 - Versions:
   - Pode: 2.6.2
   - PowerShell: 7.2.0 (Running Pode Server)

### Additional Context
Issued emerged when troubleshooting https://github.com/Badgerati/Pode/issues/977, meaning, the server declaration is more or less the same.

**Comment by Viajaz** [comment]
I tried removing the `$WebEvent.Request.AllowClientCertificate` check in the IIS Middleware, the behaviour remains, I guess that wasn't the cause of the issue. I don't think the AuthScheme is even invoking the ClientCertificate ScriptBlock

**Comment by Viajaz** [comment]
HTTP 403 still occurs even if the Pode Route has no Authentication specified:
```powershell
Add-PodeRoute -Method Get -Path '/api/v1.0/testnoauth' -ScriptBlock {
```

**Comment by Badgerati** [comment]
Hi @Viajaz,

`-AllowClientCertificate` on an HTTP endpoint should work when being hosted with IIS. That `Test-PodeIsHosted` call returns true if using IIS or Heroku, so that AllowClientCert works; if it's failing even when IIS starts the script, that sounds like IIS isn't creating the environment variables uses to detect IIS 🤔 

For the 403 even with no client cert authentication, Pode doesn't have a path to return a 403 in this instance, so it could be from IIS itself. The only times Pode actually returns a 403 is when using Bearer auth or CSRF.

Possibly some hints in #879 for IIS - if I remember you have to set `Require` for client certs rather than `Accept`.

**Comment by Viajaz** [comment]
I enabled IIS Request Tracing on the Pode site, the exact error was `403.16` which is specifically:
`A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider. (0x800b0109)`

Meaning, this was an IIS configuration issue and not a Pode issue. In my case, the Client Certificate was issued by a Trusted Root Certificate Authority but it turns out I needed to reconfigure the binding properly so SChannel would do the chain validation correctly. The issue was fixed by: https://serverfault.com/questions/1050679/how-to-use-the-client-authentication-issuers-certificate-store-on-windows-serv

1. Added the authorised Client Certificate Issuer to the _Client Authentication Issuers_ (`ClientAuthIssuer`) Local Machine Certificate Store
2. In my case, I still wanted the TLS Certificate for the IIS Site to be in the Local Machine Personal Store (Not Web Hosting):
`netsh http update sslcert hostnameport=api.example.com:443 appid="{ffffffff-ffff-ffff-ffff-ffffffffffff}" certhash=ffffffffffffffffffffffffffffffffffffffff certstorename=My sslctlstorename=ClientAuthIssuer`

This resolved the problem for me. Could be worth adding this to the documentation for Client Certificates with IIS Hosting for Pode.
@Badgerati
